packages/components/src/filters/advanced/search-filter.js

```diff
@@ -25,7 +25,14 @@ class SearchFilter extends Component {
 		this.updateLabels = this.updateLabels.bind( this );
 
 		if ( filter.value.length ) {
-			config.input.getLabels( filter.value, query ).then( this.updateLabels );
+			if ( typeof config.input.getLabels === 'function' ) {
+				config.input.getLabels( filter.value, query ).then( this.updateLabels );
+			} else {
+				const filters = filter.value.split( ',' );
+				this.state.selected = config.input.getLabels.filter( ( label ) => {
+					return filters.includes( label.id );
+				} );
```

Comment:
In writing the `getLabels` functionality I was going for this component being as agnostic as possible to the logic required to create the `this.state.selected`.

I think by putting this into a `getLabels` function, we can keep this component clean. What do you think?

Reply:
Cleaned this up a bit, and moved back to a function in the config. Thanks for the feedback here!